src/django_pg_migration_tools/operations.py

```diff
@@ -70,17 +71,22 @@ def database_forwards(
         to_state: migrations.state.ProjectState,
     ) -> None:
         self._ensure_not_in_transaction(schema_editor)
```

Comment:
`_ensure_not_in_transaction` checks an attribute of the DB connection, which presumably establishes the connection if it hasn't been opened already. If we're not intending to apply this migration to this particular DB, then we should probably avoid opening the connection. In other words, push this check into the `if` block below.

Reply:
That's a good observation. Django's `AddIndexConcurrently` behaves in the same way as in this PR. It runs the `_ensure_not_in_transaction` check [before verifying the router](https://github.com/django/django/blob/7adb6dd98d50a238f3eca8c15b16b5aec12575fd/django/contrib/postgres/operations.py#L136). That's why I decided to keep it here for consistency.

This seems to be a fail-early approach.